### DISCUSSION TITLE
How to use YT to Blog feature?

### DISCUSSION THREAD

**Discussion by XamHans** [ALwrity/ALwrity#131] [General]
Hi ALwrity team and community,

I'm new to ALwrity and interested in the YouTube video to blog post conversion feature. I've set up some API keys, but I haven't provided a YouTube API key. I have a couple of questions:

1. How exactly do I use this feature? Are there any specific steps I need to follow?
2. How does the conversion process work without a YouTube API key?
3. It would be great to see all options on the config 

Also, I just wanted to say that ALwrity is a really cool project. I actually started thinking about a similar idea today, so it's great to see it already exists!

Thanks for any help, and for building this tool!

Best,
Johannes

**Comment by AJaySi** [comment]
@XamHans 
Thank you so much Johannes, for your kind words.

You only need to provide youtube link and it should do the rest. There is no need for youtube API. We first download the audio from youtube and then pass the downloaded audio to LLM to transcribe it. Once its transcribed, we add google search result and create a similar/new blog from it.  I tried it fails with a small error. _I will fix this today._ 

```
AI-Writer/lib/gpt_providers/audio_to_text_generation$ ls
gemini_audio_text.py  __pycache__  stt_audio_blog.py
AI-Writer/lib/ai_writers/speech_to_blog$ ls
main_audio_to_blog.py  __pycache__  write_blogs_from_youtube_videos.py
```

As you are also thinking about this idea, ALwrity really needs your help.
Let me know what you would like to do and I will help you get started.

Regards,
Aj.

**Comment by AJaySi** [comment]
https://github.com/AJaySi/AI-Writer/issues/133
